tests/Unit/DataStructures/VectorImplTestHelper.hpp

```diff
@@ -160,5 +163,349 @@ auto remove_nth(const std::tuple<ValueTypes...>& tup) noexcept {
       tup, std::make_index_sequence<N>(),
       std::make_index_sequence<sizeof...(ValueTypes) - N - 1>());
 }
+
+/// \ingroup TestingFrameworkGroup
+/// \brief test construction and assignment of a `VectorType` with a `ValType`
+template <typename VectorType, typename ValType>
+void vector_test_construct_and_assign(
+    typename tt::get_fundamental_type_t<ValType> low =
+        typename tt::get_fundamental_type_t<ValType>{-100.0},
+    typename tt::get_fundamental_type_t<ValType> high =
+        typename tt::get_fundamental_type_t<ValType>{100.0}) noexcept {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<typename tt::get_fundamental_type_t<ValType>> dist{
+      low, high};
+  UniformCustomDistribution<size_t> sdist{2, 20};
+
+  const size_t size = sdist(gen);
+
+  const VectorType size_constructed{size};
+  CHECK(size_constructed.size() == size);
+  const auto generated_val1 = make_with_random_values<ValType>(
```

Comment:
I have a mild preference for the `x_val` variable names to become `x_value` to match the use of `ValueType` as the template parameter. But feel free to veto.